internal/kafka/handler.go

```diff
@@ -114,18 +115,18 @@ func (this *handler) onMessage(ctx context.Context, msg *kafka.Message, cfg *con
 	// Insert payload into DB
 	endpoints.ObserveMessageProcessTime(time.Since(start))
 	endpoints.IncMessagesProcessed()
-	result := queries.InsertPayloadStatus(this.db, sanitizedPayloadStatus)
-	if result.Error != nil {
-		l.Log.Debug("Failed to insert sanitized PayloadStatus with ERROR: ", result.Error)
-		result = queries.InsertPayloadStatus(this.db, sanitizedPayloadStatus)
-		if result.Error != nil {
-			l.Log.Debug("Failed to re-insert sanitized PayloadStatus with ERROR: ", result.Error)
-			result = queries.InsertPayloadStatus(this.db, sanitizedPayloadStatus)
-			if result.Error != nil {
-				endpoints.IncMessageProcessErrors()
-				l.Log.Error("Failed final attempt to re-insert PayloadStatus with ERROR: ", result.Error)
-			}
+
+	// TODO: Configurable retries
+	retries, attempts := 3, 0
+	for retries >= attempts {
+		if err := attemptPayloadInsertion(this.db, sanitizedPayloadStatus); err != nil {
+			l.Log.WithFields(logrus.Fields{"attempts": attempts}).Debug("Failed to insert sanitized PayloadStatus with ERROR: ", err)
+			attempts += 1
+
+			continue
 		}
+
+		break
```

Comment:
Kinda nit-picky but I think I would change this logic so that if the insert succeeds (err == nil) then jump out of the loop...if the insert fails, then just loop back around and try again.

Would it make this cleaner to move the retry logic into the `attemptPayloadInsertion()` function?  :thinking: